### PR TITLE
Add recipe for ob-ffuf

### PR DESCRIPTION
**Pull request by daniel-ts**
### Brief summary of what the package does

This package integrates the [ffuf](https://github.com/ffuf/ffuf) web fuzzer with Org Babel, so that an HTTP request may be written inside a source code block and an Orgmode table specified as a wordlist input and fired off quickly.

### Direct link to the package repository

[https://github.com/daniel-ts/ob-ffuf](https://github.com/daniel-ts/ob-ffuf)

### Your association with the package

I am the maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

### CODE REVIEW

File: recipes/ob-ffuf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+(ob-ffuf
+ :repo "daniel-ts/ob-ffuf"
+ :fetcher github
+ :files ("*.el"
+         (:exclude "*-test.el" "README.org" ".gitignore")))
```
Comment (riscy) [inline recipes/ob-ffuf:5]: Equivalent:
```suggestion
(ob-ffuf :repo "daniel-ts/ob-ffuf" :fetcher github)
```

**Review by riscy — COMMENTED** [review]
Thanks for this.  Quick first pass --

`ob-ffuf-helpers.el` with byte-compile using Emacs 28.1:
```
ob-ffuf-helpers.el:55:28: Warning: `(check-header (lambda (header) (and
    (stringp header) (not (eq header "")))))' is a malformed function
In ob-ffuf--handle-proc-events:
ob-ffuf-helpers.el:170:43: Warning: `(funcall-or-ignore (lambda (fn &rest
    args) (if fn (apply fn args) (ignore args))))' is a malformed function
In end of data:
ob-ffuf-helpers.el:212:22: Warning: the function `org-babel--expand-body' is
    not known to be defined.
ob-ffuf-helpers.el:208:17: Warning: the function `org-babel-lob--src-info' is
    not known to be defined.
ob-ffuf-helpers.el:188:15: Warning: the function `funcall-or-ignore' is not
    known to be defined.
ob-ffuf-helpers.el:168:8: Warning: the function `org-babel-ref-resolve' is not
    known to be defined.
ob-ffuf-helpers.el:81:29: Warning: the function `org-babel-temp-file' is not
    known to be defined.
ob-ffuf-helpers.el:69:31: Warning: the function `check-header' is not known to
    be defined.
ob-ffuf-helpers.el:59:4: Warning: the function `cl-flet' is not known to be
    defined.
```

`ob-ffuf-helpers.el` with package-lint 20220913.1226:
```
1 issue found:
5:0: error: Package-Requires outside the main file have no effect.
```

`ob-ffuf-helpers.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- ob-ffuf-helpers.el#L6: Prefer `;; Version` over `;; Package-Version` (MELPA automatically adds `Package-Version`)
- ob-ffuf-helpers.el#L107: `concat` may be unneeded; `insert` can take multiple arguments
```

`ob-ffuf-mode.el` with byte-compile using Emacs 28.1:
```
ob-ffuf-mode.el:25:7: Warning: assignment to free variable
    `ob-ffuf--mode-keywords'
```

`ob-ffuf-mode.el` with checkdoc 0.6.2:
```
ob-ffuf-mode.el:0: You should have a section marked ";;; Commentary:"
ob-ffuf-mode.el:25: You should have a section marked ";;; Code:"
```

`ob-ffuf-mode.el` with package-lint 20220913.1226:
```
1 issue found:
69:0: error: "ffuf-mode" doesn't start with package's prefix "ob-ffuf".
```

`ob-ffuf-mode.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- ob-ffuf-mode.el#L44: It's safer to sharp-quote function names; use `#'`
- ob-ffuf-mode.el#L53: It's safer to sharp-quote function names; use `#'`
- ob-ffuf-mode.el#L25: Top-level `setq` should usually be replaced by `defvar`
```

`ob-ffuf.el` with checkdoc 0.6.2:
```
ob-ffuf.el:51: You should have a section marked ";;; Code:"
ob-ffuf.el:124: All variables and subroutines might as well have a documentation string
ob-ffuf.el:127: All variables and subroutines might as well have a documentation string
```

`ob-ffuf.el` with package-lint 20220913.1226:
```
2 issues found:
123:0: error: "org-babel-expand-body:ffuf" doesn't start with package's prefix "ob-ffuf".
123:0: error: `org-babel-expand-body:ffuf' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
```

Package and license:
- Add a LICENSE file to the repository

**Comment by daniel-ts** [comment]
Hi. Thank you for the quick review!

I addressed the errors and warnings. The current master branch of `ob-ffuf` contains the fixes.

The following warnings cannot be fixed, because it's an Org Babel convention:
2 issues found:
123:0: error: "org-babel-expand-body:ffuf" doesn't start with package's prefix "ob-ffuf".
123:0: error: `org-babel-expand-body:ffuf' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).

**Comment by riscy** [comment]
Noted -- the prefix rules around org-babel packages trip me up every time.  Here's the latest --

`ob-ffuf-mode.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- ob-ffuf-mode.el#L30: Top-level `setq` should usually be replaced by `defvar`
```
In fact for this case you can just merge the `defvar` and the `setq` into a single form.

Taking a deeper dive into the code:
- `(error "Malformed table")` in ob-fuff-helpers.el; should this be a `user-error` that doesn't open up a backtrace?
- ob-fuff.el: `Babel funcitons for ffuf` functions should be `functions` (this one is important because this summary line will appear foremost in the MELPA index)
- ob-ffuf-helpers.el: funcitons -> functions

Looks like a quick merge after these.

**Comment by daniel-ts** [comment]
Hi. Thank you again for further feedback.

I fixed the mentioned issues.

One question: Will changes to the code need approval or will new commits to the ``ob-ffuf`` repo be available as updates on melpa?

**Comment by riscy** [comment]
> Will changes to the code need approval or will new commits to the ob-ffuf repo be available as updates on melpa?

Updates will be available automatically and won't need further approval.  MELPA have discussed the possibility of crawling existing packages and displaying a score in the MELPA index, but that's not anything that's been implemented.

Merging now - welcome to MELPA. :)

**Pull request merged**